main.go

```diff
@@ -106,6 +111,30 @@ func main() {
 			return ""
 		}
 	}
+	if soFiles, err := filepath.Glob(path.Join(*plugDir, "*.so")); err != nil {
+		log.Printf("could not search for plugins in directory %q: %v", *plugDir, err)
+	} else {
+		for _, so := range soFiles {
+			plug, err := plugin.Open(so)
+			if err != nil {
+				log.Printf("could not open plugin %q: %v", so, err)
+				continue
+			}
+			sym, err := plug.Lookup("FuncMap")
+			if err != nil {
+				log.Printf("could not lookup symbol 'FuncMap' in plugin %q: %v", so, err)
+				continue
+			}
+			f, ok := sym.(func() template.FuncMap)
+			if !ok {
+				log.Printf("could not assert symbol 'FuncMap' to be of type func() template.FuncMap in plugin %q: %v", so, err)
+				continue
+			}
+			for k, v := range f() {
+				fm[k] = v
```

Comment:
Initially, I thought overriding functions would be fine (and possibly useful/desirable), but I can see how users might find this behavior surprising.

Returning an error might be too restrictive, however, as that would require all function names to be unique across all plugins. Since plugins might not come from the same source, that's impossible to gaurentee.

Perhaps plugin functions should be namespaced using `<plugin_name>_` as the prefix, where `<plugin_name>` is the name of the binary?